src/postgraphile/http/subscriptions.ts

```diff
@@ -271,46 +256,35 @@ export async function enhanceHttpServerWithSubscriptions<
           options,
           req,
           res,
-          variables: params.variables,
-          operationName: params.operationName,
+          variables: args.variableValues,
+          operationName: args.operationName,
           meta,
         });
         if (moreValidationRules.length) {
           const validationErrors: ReadonlyArray<GraphQLError> = validate(
-            params.schema,
-            finalParams.query,
+            args.schema,
+            typeof hookedParams.document === 'string'
+              ? parse(hookedParams.document)
+              : hookedParams.document,
             moreValidationRules,
           );
           if (validationErrors.length) {
             const error = new Error(
               'Query validation failed: \n' + validationErrors.map(e => e.message).join('\n'),
             );
             error['errors'] = validationErrors;
-            return Promise.reject(error);
+            throw error;
           }
         }
 
-        return finalParams;
+        return [hookedParams, executionResultFormatter] as [
+          ExecutionArgs,
+          ExecutionResultFormatter,
+        ];
       },
-      onOperationComplete(socket: WebSocket, opId: string) {
-        releaseContextForSocketAndOpId(socket, opId);
+      onComplete: ({ socket }, msg) => {
+        releaseContextForSocketAndOpId(socket, msg.id);
       },
-
-      /*
-       * Heroku times out after 55s:
-       *   https://devcenter.heroku.com/articles/error-codes#h15-idle-connection
-       *
-       * The subscriptions-transport-ws client times out by default 30s after last keepalive:
-       *   https://github.com/apollographql/subscriptions-transport-ws/blob/52758bfba6190169a28078ecbafd2e457a2ff7a8/src/defaults.ts#L1
-       *
-       * GraphQL Playground times out after 20s:
-       *   https://github.com/prisma/graphql-playground/blob/fa91e1b6d0488e6b5563d8b472682fe728ee0431/packages/graphql-playground-react/src/state/sessions/fetchingSagas.ts#L81
-       *
-       * Pick a number under these ceilings.
-       */
-      keepAlive: 15000,
```

Comment:
Where's the keepAlive configured now?

Reply:
I have envisioned using the [WebSocket Ping and Pong](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#Pings_and_Pongs_The_Heartbeat_of_WebSockets) for keeping the connection alive. Not only that, but my idea is to abstract away the need of managing the `keepAlive` completely and let a baked-in implementation handle that for the user.

Reply:
Different clients/networks have different needs here; so I think it definitely needs to be configurable. Using the built in websocket ping/pong makes sense to me, but I'd have to test it to make sure it works for Heroku/etc.

Reply:
Fair point, I'll make the `keepAlive` configurable. [PR #10 at the lib repo](https://github.com/enisdenjo/graphql-transport-ws/issues/10).

Reply:
Done in [PR #11 at the lib repo](https://github.com/enisdenjo/graphql-transport-ws/pull/11). Will resolve this conversation once I have released a new version and updated this fork.